IQKeyBoardManager/Categories/IQUIView+Hierarchy.m

```diff
@@ -198,6 +205,16 @@ - (NSArray*)deepResponderViews
     return textFields;
 }
 
+- (UIView *)parentView
+{
+    UIView *parentView = self;
+    while (![parentView.superview isKindOfClass:[UIWindow class]]) {
```

Comment:
If there is no top level window present, then this view is not being drawn on screen and therefore should not be the current responder. 

> On Jan 5, 2016, at 8:41 AM, Mohd Iftekhar Qurashi <notifications@github.com> wrote:
> 
> In IQKeyBoardManager/Categories/IQUIView+Hierarchy.m:
> 
> > @@ -198,6 +205,16 @@ - (NSArray*)deepResponderViews
> >      return textFields;
> >  }
> >  
> > +- (UIView *)parentView
> > +{
> > +    UIView *parentView = self;
> > +    while (![parentView.superview isKindOfClass:[UIWindow class]]) {
> When there will be no window for the view. I think the while loop never ends. I think you should also put a condition to check if parentView.superview != nil
> 
> —
> Reply to this email directly or view it on GitHub.
>